libbindgen/src/ir/comp.rs

```diff
@@ -506,6 +512,8 @@ impl CompInfo {
         debug!("CompInfo::from_ty({:?}, {:?})", kind, cursor);
 
         let mut ci = CompInfo::new(kind);
+        ci.is_forward_declaration = location.map_or(true, |cur|
+                                        cur.is_definition() && cur.kind() == CXCursor_StructDecl);
```

Comment:
Why only if the cursor is `CXCursor_StructDecl`?

Also, I think this should work ok if you just do:

```rust
ci.is_forward_declaration = !cursor.is_definition();
```

That should make it work for structs with template parameters too. Does that work?